web/pandas/pdeps/0006-ban-upcasting.md

```diff
@@ -51,65 +51,71 @@ In[11]: ser[2] = "2000-01-04x"  # typo - but pandas does not error, it upcasts t
-- setitem indexing operations (where ``indexer`` could be a slice, a mask,
+* setitem-like operations:
+
+    - ``ser.fillna('foo', inplace=True)``
+    - ``ser.where(ser.isna(), 'foo', inplace=True)``
+    - ``ser.fillna('foo', inplace=False)``
+    - ``ser.where(ser.isna(), 'foo', inplace=False)``
+
+* setitem indexing operations (where ``indexer`` could be a slice, a mask,
   a single value, a list or array of values, or any other allowed indexer):
-  - ``ser.iloc[indexer] = 'foo'``
-  - ``ser.loc[indexer] = 'foo'``
-  - ``df.iloc[indexer, 0] = 'foo'``
-  - ``df.loc[indexer, 'a'] = 'foo'``
-  - ``ser[indexer] = 'foo'``
+
+    - ``ser.iloc[indexer] = 'foo'``
+    - ``ser.loc[indexer] = 'foo'``
+    - ``df.iloc[indexer, 0] = 'foo'``
+    - ``df.loc[indexer, 'a'] = 'foo'``
+    - ``ser[indexer] = 'foo'``
 
 It may be desirable to expand the top list to ``Series.replace`` and ``Series.update``,
 but to keep the scope of the PDEP down, they are excluded for now.
 
 Examples of operations which would not raise are:
-- ``ser.diff()``;
-- ``pd.concat([ser, ser.astype(object)])``;
-- ``ser.mean()``;
-- ``ser[0] = 3``;  # same dtype
-- ``ser[0] = 3.``;  # 3.0 is a 'round' float and so compatible with 'int64' dtype
-- ``df['a'] = pd.date_range(datetime(2020, 1, 1), periods=3)``;
-- ``df.index.intersection(ser.index)``.
+
+- ``ser.diff()``
+- ``pd.concat([ser, ser.astype(object)])``
+- ``ser.mean()``
+- ``ser[0] = 3``  # same dtype
+- ``ser[0] = 3.``  # 3.0 is a 'round' float and so compatible with 'int64' dtype
+- ``df['a'] = pd.date_range(datetime(2020, 1, 1), periods=3)``
+- ``df.index.intersection(ser.index)``
 
 ## Detailed description
 
 Concretely, the suggestion is:
-- if a ``Series`` is of a given dtype, then a ``setitem``-like operation should not change its dtype;
-- if a ``setitem``-like operation would previously have changed a ``Series``' dtype, it would now raise.
+
+- If a ``Series`` is of a given dtype, then a ``setitem``-like operation should not change its dtype.
+- If a ``setitem``-like operation would previously have changed a ``Series``' dtype, it would now raise.
 
 For a start, this would involve:
 
-1. changing ``Block.setitem`` such that it does not have an ``except`` block in
+1. changing ``Block.setitem`` such that it does not have an ``except`` block in:
 
-  ```python
-  value = extract_array(value, extract_numpy=True)
-  try:
-      casted = np_can_hold_element(values.dtype, value)
-  except LossySetitemError:
-      # current dtype cannot store value, coerce to common dtype
-      nb = self.coerce_to_target_dtype(value)
-      return nb.setitem(indexer, value)
-  else:
-  ```
+    ```python
+    value = extract_array(value, extract_numpy=True)
+    try:
+        casted = np_can_hold_element(values.dtype, value)
+    except LossySetitemError:
+        # current dtype cannot store value, coerce to common dtype
+        nb = self.coerce_to_target_dtype(value)
+        return nb.setitem(indexer, value)
+    else:
+    ```
```

Comment:
this one may not be rendering correctly

Reply:
As of [e617561](https://github.com/pandas-dev/pandas/pull/53442/commits/e6175619d185c7548f30f556f09f828855c9314b) the code block within the ordered list is rendered correctly and does not break the ordered list according to local build. Code check in progress.